test/integration/parachain/staging/sequential/redeem.test.ts

```diff
@@ -116,6 +118,44 @@ describe("redeem", () => {
         );
     });
 
+    it("should have applied oracle fee rate to redeem transaction", async () => {
+        const oracleBtcFeePerByte = await interBtcAPI.oracle.getBitcoinFees();
+
+        const redeemRequests = await interBtcAPI.redeem.list();
+        for (const redeemRequest of redeemRequests) {
+            const opreturnData = stripHexPrefix(redeemRequest.id.toString());
+            const txId = await interBtcAPI.electrsAPI.getTxIdByOpReturn(opreturnData)
+                .catch((e) => {
+                    // if no opreturn is ready, we may still have to wait for it
+                    return interBtcAPI.electrsAPI.waitForOpreturn(opreturnData, 5 * 60 * 1000, 5000);
+                });
+            if (txId === undefined) {
+                assert.fail(`Could not fetch BTC transaction id for redeem request id ${redeemRequest.id}`);
+                return;
+            }
+
+            // get the actual values on the BTC transaction
+            const btcTx = await interBtcAPI.electrsAPI.getTx(txId);
+
+            const actualTxFeeSatoshi = new Big(btcTx.fee || 0);
+            const actualTxWeight = new Big(btcTx.weight || 0);
+            const actualTxVsize = actualTxWeight.div(4).round(0, RoundingMode.RoundUp);
```

Comment:
Would be useful to add a comment explaining that Vsize is equal to 4 weight units and that is why we divide. Could even be a util function.